### PR TITLE
Add big sky button to goal centric design picker

### PR DESCRIPTION
**Pull request by madhusudhand**
Related to: https://github.com/Automattic/dotcom-forge/issues/10011

## Proposed Changes

* Adds new button `Create yours with AI` to the design picker for eligible site plans.
* Adds tracking events

image

## Why are these changes being made?

* To enhance wordpress on-boarding experience.

## Testing Instructions

* Create a site with Premium or Business plans
* Go to design picker (on-boarding step)
* Click on the button `Create your with AI` to start designing the site with BigSky

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx [hunk 1]
```diff
@@ -962,6 +966,35 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
...
+		// is_filter_included_with_plan_enabled: true/false,
+		// preselected_filters: ??,
+		// selected_filters: ??,
+		// {filter} ??
+	};
+	const bigSkyButtons = (
+		<>
+			{ isBigSkyEligible && (
+				<Button
+					href={ `/setup/site-setup/launch-big-sky?siteSlug=${ siteSlug }&siteId=${ site.ID }` }
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: I suppose we can simply use `navigate( 'launch-big-sky' )` to go to the big sky step 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx [hunk 2]
```diff
@@ -962,6 +966,35 @@ const UnifiedDesignPickerStep: Step = ( { navigation, flow, stepName } ) => {
...
 	);
 
+	const bigSkyButtonEventProperties = {
+		is_big_sky_eligible: isBigSkyEligible,
+		// is_filter_included_with_plan_enabled: true/false,
+		// preselected_filters: ??,
+		// selected_filters: ??,
+		// {filter} ??
+	};
+	const bigSkyButtons = (
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:979]: Let's gate the feature like [here](https://github.com/Automattic/wp-calypso/pull/97004/files#diff-a539d272b24cf98cb319fb27637a06c9faf52cfea40d0af87cf1b7c2fec0915cR122) and add the common properties in the follow-up PR!

Comment (madhusudhand) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx:979]: Updated here e47aad3afe330247945a0fa0cea6141480b05cbc

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx (hunk 1 above)
Comment (madhusudhand) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: `navigate('launch-big-sky')` is ignoring query params and making the route invalid. 
Used complete path, now it doesn't cause complete app reload. 👍 
`navigate( `/setup/site-setup/launch-big-sky?siteSlug=${ siteSlug }&siteId=${ site.ID }` );`

**Review by arthur791004 — APPROVED** [review]
👍

**Comment by taipeicoder** [comment]
@madhusudhand could you add this PR to the team board?

**Comment by dognose24** [comment]
HI, @madhusudhand. This [const for checking feature flag](https://github.com/Automattic/wp-calypso/pull/97006/files#diff-a539d272b24cf98cb319fb27637a06c9faf52cfea40d0af87cf1b7c2fec0915cR465) is duplicated and causes a compile error on the development console.

截圖 2024-12-05 下午3 01 04

**Comment by arthur791004** [comment]
@dognose24 It's fixed by https://github.com/Automattic/wp-calypso/pull/97099

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17047127

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @madhusudhand for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17047127).

**Pull request merged**